src/App.tsx

```diff
@@ -1,15 +1,51 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [chosenPerson, setChosenPerson] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const handleQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!query && !e.target.value.trim()) {
+      return;
+    }
```

Comment:
The condition in this if statement doesn't meet the requirement from the checklist: 'make sure that filter won't be called if user entered spaces only'. The current condition only checks if the query is empty, but it doesn't handle the case when the user enters spaces only. Consider adding an additional check for this case.